File: app/src/main/java/com/evanzheng/bibliobarcode/BookDao.java
package com.evanzheng.bibliobarcode;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

//A DAO that manages queries to our SQL database
@Dao
public interface BookDao {
    @Insert
    void insertBook(Book book);

    @Update
    void updateBook(Book book);

    @Query("DELETE FROM books WHERE isbn=:isbn")
    void deleteBook(String isbn);

    @Query("SELECT isbn FROM books")
    List<String> loadISBN();

    @Query("SELECT * FROM books")
    List<Book> loadBookSources();

    @Query("SELECT * FROM books WHERE isbn =:isbn")
    Book loadBook(String isbn);
}

File: app/src/main/java/com/evanzheng/bibliobarcode/BookDao.kt
package com.evanzheng.bibliobarcode

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

//A DAO that manages queries to our SQL database
@Dao
interface BookDao {
    //Inserting a book into the database
    @Insert
    fun insertBook(book: Book?)

    //Updating an existing book in the database
    @Update
    fun updateBook(book: Book?)

    //Deleting a book with a certain ISBN
    @Query("DELETE FROM books WHERE isbn=:isbn")
    fun deleteBook(isbn: String?)

    //Getting list of all ISBNs currently in database
    @Query("SELECT isbn FROM books")
    fun loadISBN(): List<String?>?

    //Getting list of all books currently in database
    @Query("SELECT * FROM books")
    fun loadBookSources(): List<Book?>?

    //Selecting a book with a particular ISBN from database
    @Query("SELECT * FROM books WHERE isbn =:isbn")
    fun loadBook(isbn: String?): Book?
}